src/OrchardCore.Modules/OrchardCore.Media/Deployment/MediaDeploymentStepRender.cs

```diff
@@ -0,0 +1,60 @@
+using System.Linq;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using OrchardCore.Media.ViewModels;
+
+namespace OrchardCore.Media.Deployment
+{
+    public static class MediaDeploymentStepRender
+    {
+        public static IHtmlContent RenderMediaStoreEntries(this IHtmlHelper<MediaDeploymentStepViewModel> helper, MediaDeploymentStepViewModel model)
+        {
+            var contentBuilder = new HtmlContentBuilder();
+
+            contentBuilder.AppendHtmlLine("<ul class=\"list-group\">");
```

Comment:
What is the reason to render some UI from a class?

Reply:
Because this is recursive ... at line 45 the function calls back upon itself. I struggled with doing this in the view itself in a function ... I'll give it one more shot.

>             What is the reason to render some UI from a class?

Reply:
I am struggling with this being a relative newcomer to ASP.NET Core and I know this is very basic stuff. I added a partial view but the app keeps throwing an exception saying it cannot find the partial view.

I have tried placing it in the Views folder, in Views/Shared, in Views/Step. I have tried naming with an underscore prefix and without. None of the combinations seem to work.

Do I need to add an area route or place the partial view in a specially named folder?

I consistently get this error:

InvalidOperationException: The partial view 'MediaDeploymentStep.Entries' was not found. The following locations were searched:
/Areas/TheAdmin/Views/OrchardCore.Deployment/Step/MediaDeploymentStep.Entries.cshtml
/Areas/TheAdmin/Views/OrchardCore.Deployment/Shared/MediaDeploymentStep.Entries.cshtml
/Areas/TheAdmin/Views/Shared/MediaDeploymentStep.Entries.cshtml
/Areas/OrchardCore.Deployment/Views/Step/MediaDeploymentStep.Entries.en-US.cshtml
/Areas/OrchardCore.Deployment/Views/Step/MediaDeploymentStep.Entries.en.cshtml
/Areas/OrchardCore.Deployment/Views/Step/MediaDeploymentStep.Entries.cshtml
/Areas/OrchardCore.Deployment/Views/Shared/MediaDeploymentStep.Entries.en-US.cshtml
/Areas/OrchardCore.Deployment/Views/Shared/MediaDeploymentStep.Entries.en.cshtml
/Areas/OrchardCore.Deployment/Views/Shared/MediaDeploymentStep.Entries.cshtml
/Views/Shared/MediaDeploymentStep.Entries.en-US.cshtml
/Views/Shared/MediaDeploymentStep.Entries.en.cshtml
/Views/Shared/MediaDeploymentStep.Entries.cshtml
/Pages/Shared/MediaDeploymentStep.Entries.en-US.cshtml
/Pages/Shared/MediaDeploymentStep.Entries.en.cshtml
/Pages/Shared/MediaDeploymentStep.Entries.cshtml

Any suggestions?